### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by kimberscott**

### CODE REVIEW

File: exp/views/study.py [hunk 1]
```diff
@@ -565,9 +565,7 @@ def get_context_data(self, **kwargs):
         are paginated.
         """
         context = super().get_context_data(**kwargs)
-        responses = context['study'].responses.order_by('id')
-        context['all_responses'] = ', '.join(self.build_responses(responses))
-        context['csv_responses'] = self.build_all_csv(responses)
+        context['n_responses'] = len(context['study'].responses.order_by('id'))
```
Comment (pattisdr) [inline exp/views/study.py]: No need to order these by `id`

File: exp/views/study.py [hunk 2]
```diff
@@ -621,9 +619,7 @@ def get_context_data(self, **kwargs):
         are paginated.
         """
         context = super().get_context_data(**kwargs)
-        responses = context['study'].responses.order_by('id')
-        context['all_responses'] = ', '.join(self.build_participant_data(responses))
-        context['csv_responses'] = self.build_all_participant_csv(responses)
+        context['n_responses'] = len(context['study'].responses.order_by('id'))
```
Comment (pattisdr) [inline exp/views/study.py]: No need to order these by `id`.

File: studies/templates/studies/_all_json_and_csv_data.html [hunk 3]
```diff
@@ -71,22 +67,28 @@ <h1>All Responses</h1>
...
                     </a>
                     <a id='download-all-data-csv' class='hidden btn btn-primary' href="{% url 'exp:study-responses-download-csv' pk=study.id %}">
-                        Download All Responses as CSV</span>
+                        Download all {{ n_responses }} responses as CSV</span>
                     </a>
                 {% endif %}
                 {% if active == 'demographics' %}
                     <a id='download-all-data-json' class ='btn btn-primary' href="{% url 'exp:study-demographics-download-json' pk=study.id %}">
-                        Download All Demographic Snapshots as JSON</span>
+                        Download all {{ n_responses }} demographic snapshots as JSON</span>
```
Comment (pattisdr) [inline studies/templates/studies/_all_json_and_csv_data.html]: There's a django template `pluralize` helper you can use, for the case when there's one snapshot or response https://docs.djangoproject.com/en/2.0/ref/templates/builtins/#pluralize

Comment (kimberscott) [reply studies/templates/studies/_all_json_and_csv_data.html]: Oh cool, thank you!

File: exp/views/study.py (hunk 1 above)
Comment (kimberscott) [reply exp/views/study.py]: Whoops, thanks for catching that.

File: web/templates/registration/password_reset_form.html [hunk 4]
```diff
@@ -1,29 +1,33 @@
...
-        <label for="id_email">{% trans 'Email address:' %}</label>
-        {{ form.email }}
+            <form method="post">{% csrf_token %}
+			<fieldset class="module aligned">
+				<div class="form-row field-email">
+					{{ form.email.errors }}
+					<label for="id_email">{% trans 'Email address:' %}</label>
+					{{ form.email }}
+				</div>
+                <input type="submit" class="btn btn-success" value="{% trans 'Reset my password' %}" />
```
[the remark this answers is not in the document]
Comment (kimberscott) [reply web/templates/registration/password_reset_form.html]: Added some space to these.

**Pull request merged**